pkg/vendir/fetch/helmchart/sync.go

```diff
@@ -63,21 +61,16 @@ func (t *Sync) Sync(dstPath string, tempArea ctlfetch.TempArea) (ctlconf.LockDir
 
 	defer os.RemoveAll(chartsDir)
 
-	helmHomeDir, err := tempArea.NewTempDir("helm-home")
-	if err != nil {
-		return lockConf, err
-	}
-
-	defer os.RemoveAll(helmHomeDir)
-
-	err = t.init(helmHomeDir)
-	if err != nil {
-		return lockConf, err
-	}
-
-	err = t.fetch(helmHomeDir, chartsDir)
-	if err != nil {
-		return lockConf, err
+	if t.opts.Repository != nil && strings.HasPrefix(t.opts.Repository.URL, "oci://") {
+		err = NewOCISource(t.opts, t.helmBinary, t.refFetcher).Fetch(chartsDir, tempArea)
+		if err != nil {
+			return lockConf, err
+		}
+	} else {
+		err = NewHTTPSource(t.opts, t.helmBinary, t.refFetcher).Fetch(chartsDir, tempArea)
+		if err != nil {
+			return lockConf, err
+		}
```

Comment:
Could move `if err != nil` check out side if-else blocks to avoid duplication, but might make sense to keep for readability.

Reply:
readability is what i would favor in this case. this duplication isnt a business related logic duplication so wouldnt lead to complications.